src/SweetAlert.js

```diff
@@ -73,191 +73,195 @@ function warningRemoved(props) {
+      text: PropTypes.string,
+      type: PropTypes.oneOf(['warning', 'error', 'success', 'info', 'input']),
+      customClass: PropTypes.string,
+      showCancelButton: PropTypes.bool,
+      showConfirmButton: PropTypes.bool,
+      confirmButtonText: PropTypes.string,
+      confirmButtonColor: PropTypes.string,
+      confirmButtonClass: PropTypes.string,
+      cancelButtonText: PropTypes.string,
+      cancelButtonClass: PropTypes.string,
+      reverseButtons: PropTypes.bool,
+      buttonsStyling: PropTypes.bool,
+      imageUrl: PropTypes.string,
+      html: PropTypes.bool,
+      animation: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(['pop', 'slide-from-top', 'slide-from-bottom']),
+      ]),
+      // inputType: PropTypes.oneOf(ALLOWS_INPUT_TYPES),
+      inputPlaceholder: PropTypes.string,
+      inputValue: PropTypes.string,
+      showLoaderOnConfirm: PropTypes.bool,
 
-    // custom option
-    show: PropTypes.bool,
-    onConfirm: PropTypes.func,
-    onCancel: PropTypes.func,
-    onClose: PropTypes.func,
-    onEscapeKey: PropTypes.func,
-    onOutsideClick: PropTypes.func,
-  }
-  /* eslint-enable react/no-unused-prop-types */
+      // custom option
+      show: PropTypes.bool,
+      onConfirm: PropTypes.func,
+      onCancel: PropTypes.func,
+      onClose: PropTypes.func,
+      onEscapeKey: PropTypes.func,
+      onOutsideClick: PropTypes.func,
+    }
+    /* eslint-enable react/no-unused-prop-types */
 
-  static defaultProps = {
-    // sweetalert option
-    text: null,
-    type: null,
-    customClass: null,
-    showCancelButton: false,
-    showConfirmButton: true,
-    confirmButtonText: 'OK',
-    confirmButtonColor: '#aedef4',
-    cancelButtonText: 'Cancel',
-    cancelButtonClass: null,
-    confirmButtonClass: null,
-    buttonsStyling: true,
-    reverseButtons: false,
-    imageUrl: null,
-    html: false,
-    animation: true,
-    // inputType: 'text',
-    inputPlaceholder: null,
-    inputValue: null,
-    showLoaderOnConfirm: false,
+    static defaultProps = {
+      // sweetalert option
+      text: null,
+      type: null,
+      customClass: null,
+      showCancelButton: false,
+      showConfirmButton: true,
+      confirmButtonText: 'OK',
+      confirmButtonColor: '#aedef4',
+      cancelButtonText: 'Cancel',
+      cancelButtonClass: null,
+      confirmButtonClass: null,
+      buttonsStyling: true,
+      reverseButtons: false,
+      imageUrl: null,
+      html: false,
+      animation: true,
+      // inputType: 'text',
+      inputPlaceholder: null,
+      inputValue: null,
+      showLoaderOnConfirm: false,
 
-    // custom option
-    show: false,
-  }
+      // custom option
+      show: false,
+    }
 
-  constructor(props, context) {
-    super(props, context)
+    constructor(props, context) {
+      super(props, context)
 
-    this._show = false
-    this.swal = props.swal || swal
-  }
+      this._show = false
+      this._swal = swalInstance
```

Comment:
The previous merge with `this.swal = props.swal || swal` was not published, so I think we can accept this as much cleaner solution with two exports